subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -155,33 +160,40 @@ static int key_gen_public_key(struct bt_conn *conn, struct fp_keys_keygen_params
 	return err;
 }
 
-static int key_gen_account_key(struct bt_conn *conn, struct fp_keys_keygen_params *keygen_params)
+static bool key_gen_account_key_check(const uint8_t *account_key, void *context)
 {
 	int err;
+	uint8_t req[FP_CRYPTO_AES128_BLOCK_LEN];
+	struct bt_conn *conn =
+		((struct fp_key_gen_account_key_check_context *) context)->conn;
+	struct fp_keys_keygen_params *keygen_params =
+		((struct fp_key_gen_account_key_check_context *) context)->keygen_params;
 	struct fp_procedure *proc = &fp_procedures[bt_conn_index(conn)];
 
-	uint8_t req[FP_CRYPTO_AES128_BLOCK_LEN];
-	uint8_t ak[CONFIG_BT_FAST_PAIR_STORAGE_ACCOUNT_KEY_MAX][FP_CRYPTO_ACCOUNT_KEY_LEN];
-	size_t ak_cnt = CONFIG_BT_FAST_PAIR_STORAGE_ACCOUNT_KEY_MAX;
+	memcpy(proc->aes_key, account_key, FP_CRYPTO_ACCOUNT_KEY_LEN);
 
-	err = fp_storage_account_keys_get(ak, &ak_cnt);
+	err = fp_keys_decrypt(conn, req, keygen_params->req_enc);
 	if (err) {
-		return err;
+		return false;
 	}
 
-	for (size_t i = 0; i < ak_cnt; i++) {
-		memcpy(proc->aes_key, ak[i], FP_CRYPTO_ACCOUNT_KEY_LEN);
+	err = keygen_params->req_validate_cb(conn, req, keygen_params->context);
+	if (err) {
+		return false;
+	}
 
-		err = fp_keys_decrypt(conn, req, keygen_params->req_enc);
-		if (!err) {
-			err = keygen_params->req_validate_cb(conn, req, keygen_params->context);
-		}
+	return true;
```

Comment:
Just a suggestion
You could have one exit point in this function and the end to be consistent with other static function in this module.

```suggestion
        if (!err) {
	    err = keygen_params->req_validate_cb(conn, req, keygen_params->context);
	}

	return (err == 0) ? true : false;
```